### DISCUSSION TITLE
Will I be able to use ecore models / metamodels?

### DISCUSSION THREAD

**Discussion by magick93** [eclipse-langium/langium#397] [Q&A]
Hello

I'm interested in langium. It looks very promising!

Quick question - will I be able to generate artifacts (eg, similar to what xtend offers) or create a dsl from an existing ecore model?

Thanks

**Comment by spoenemann** [comment]
Hi! It seems these are two separate questions:

> will I be able to generate artifacts (eg, similar to what xtend offers)

Yes, you can use template strings that are part of TypeScript or the `GeneratorNode` API offered by Langium. Usually generators are packaged in a CLI, see our statemachine example:
https://github.com/langium/langium/blob/main/examples/statemachine/src/cli/generator.ts

A very simple "Hello world" generator is also included in the projects generated by [generator-langium](https://www.npmjs.com/package/generator-langium).

> create a dsl from an existing ecore model?

Langium will not support EMF models. However, we might implement a utility to help migrating from Xtext / EMF to Langium in the future.

**Reply by magick93** [reply]
Ok, I see.

Its a shame that there will not be support for EMF models. But a migration path would be great.

Regarding templating - so will a custom template be possible or will it need to be bundled with the cli?

**Reply by spoenemann** [reply]
What do you mean by a custom template?
